File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_amap.cpp
#include <iostream>
#include <map>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so canh: ";
    cin >> m;

    // Su dung map de luu AMap
    map<int, vector<pair<int, int>>> adjMap;

    cout << "Nhap danh sach ke mo rong (u v w):\n";
    for (int i = 0; i < m; ++i) {
        int u, v, w;
        cin >> u >> v >> w;
        adjMap[u].push_back({v, w});
    }

    cout << "Adjacency Map (AMap):\n";
    for (int u = 0; u < n; ++u) {
        cout << u << ": ";
        if (adjMap.count(u)) {
            for (auto [v, w] : adjMap[u]) {
                cout << "(" << v << ", " << w << ") ";
            }
        }
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_amap.py
n = int(input("Nhap so dinh: "))
m = int(input("Nhap so canh: "))

# Khoi tao AMap: su dung dictionary
adj_map = {}

print("Nhap danh sach ke mo rong (u v w):")
for _ in range(m):
    u, v, w = map(int, input().split())
    if u not in adj_map:
        adj_map[u] = []
    adj_map[u].append((v, w))  # Them (v, w) vao danh sach cua u

print("Adjacency Map (AMap):")
for u in range(n):
    print(f"{u}:", adj_map.get(u, []))  # Neu khong co dinh u thi tra ve []